course1/module7/Question1.js

```diff
@@ -0,0 +1,32 @@
+/*1.
+ How will you create a new copy of the object below while updating the value of address.details[0] to “5“?
+{
+ name:”Harry Potter”,
+ age: 12,
+ address: {
+  details: [“4”, “Privet Drive”],
+  area:”Little Whinging”,
+  city: “Surrey”,
+  state: “England”
+ } 
+} */
+
+let emp = {
+    name:"Harry Potter",
+    age: 12,
+    address: {
+     details: ["4", "Privet Drive"],
+     area:"Little Whinging",
+     city: "Surrey",
+     state: "England"
+    } 
+}
+
+
+ const clone = JSON.parse(JSON.stringify(emp));
```

Comment:
Right. One way to clone is using Json stringify & parse but there are other ways aswell
Can you do this using spread operator?

Reply:
No,because it is a nested object so address will have same reference, creating a shallow copy.